src/Model/ModelManagerInterface.php

```diff
@@ -21,6 +21,8 @@
 
 /**
  * A model manager is a bridge between the model classes and the admin functionality.
+ *
+ * @psalm-template T
```

Comment:
can you use phpstan instead? thats what I am using on master

Reply:
Also you should look like at the recent merged pr on master, all of this is added there, the work should be backported

Reply:
I'll do a backport

Reply:
> can you use phpstan instead? thats what I am using on master

phpstan does not provide advanced annotations like `class-string`

Reply:
👍 I am doing a Narrow the Api on master branch, and in general it is a bc break. Some parts could be bc compat but spliting the already hard work makes it super hard. So if someone wants to backport the bc compat parts I am happy with it, less conflicts